[ENH] Add example config for deploying a node behind NGINX and clarify data uploading (#240)
neurobagel/documentation: docs/config.md

- [acme-companion](https://github.com/nginx-proxy/acme-companion) to automatically create and renew SSL certificates for NGINX proxied service containers

??? abstract "Example `docker-compose.yml` with NGINX reverse proxy"
    To use this file:

    1. If you haven't already, follow the [steps](getting_started.md#the-neurobagel-node-deployment-recipe) to clone and minimally configure the services in the [Neurobagel deployment recipe](https://github.com/neurobagel/recipes)
    2. Replace the default `docker-compose.yml` in the `recipes` directory with the below file
    2. Open the file, and edit the values of `VIRTUAL_HOST` and `LETSENCRYPT_HOST` to the domains your proxied services will use (see  :material-plus-circle:)
        - This assumes you have already registered your domain(s) with a DNS provider and configured the DNS settings to resolve correctly to your host machine

    3. From the root of the `recipes` directory, run:
        ```bash
        docker compose up -d
        ```
        (or, see [here](#available-profiles) to launch a non-default service profile)

>     4. Ensure that ports 80 and 443 are reachable for the host machine where your Docker Compose stack is running

Reviewer: ```suggestion
    4. Ensure that ports 80 and 443 are open on the host machine where your Docker Compose stack is running
```
